### PR TITLE
scheduler: reduce memory used in median filter

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: Ref #5692

When there are many stores and frequent flow region update, hot statistic and hot scheduler cost a lot of ram.

### What is changed and how does it work?

1. Reduce unnecessary clone in `summaryStoresLoadByEngine`, we don't need to clone all info in hot peer stat.
2. Reduce unnecessary calculate and slice in `MedianFilter`.

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
```
tiup playground nightly --kv=1 --kv.binpath=bin/pd-heartbeat-bench --kv.config=tools/pd-heartbeat-bench/config-template.toml --pd=1 --pd.binpath=bin/pd-server --db=0 --host xx.xx.xx.xx
```
notes: tests should be the same duration

Before this pr
image

After this pr

image

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by rleungx** [comment]
CI failed

File: pkg/movingaverage/median_filter.go [hunk 1]
```diff
@@ -14,57 +14,77 @@
...
+	"sync"
+
+	"github.com/montanaflynn/stats"
+)
 
 // MedianFilter works as a median filter with specified window size.
 // There are at most `size` data points for calculating.
 // References: https://en.wikipedia.org/wiki/Median_filter.
 type MedianFilter struct {
+	sync.RWMutex
```
Comment (rleungx) [inline pkg/movingaverage/median_filter.go:27]: Will it consume more resources?

Comment (lhy1024) [reply pkg/movingaverage/median_filter.go:27]: avoid potential data race

**Comment by rleungx** [comment]
I think we still need to add a benchmark.

File: server/statistics/hot_peer.go [hunk 2]
```diff
@@ -182,13 +182,30 @@ func (stat *HotPeerStat) GetThresholds() []float64 {
 
 // Clone clones the HotPeerStat.
 func (stat *HotPeerStat) Clone() *HotPeerStat {
-	ret := *stat
```
Comment (HunDunDM) [inline server/statistics/hot_peer.go:185]: Why not use `ret := *stat`?

**Review by HunDunDM — COMMENTED** [review]
rest LGTM

**Review by rleungx — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**